src/test/java/service/TagServiceTest.java

```diff
@@ -1,83 +1,66 @@
 package service;
 
+import static com.epam.esm.exceptions.Codes.TAG_BAD_REQUEST;
+import static com.epam.esm.exceptions.Codes.TAG_NOT_FOUND;
+import static com.epam.esm.exceptions.Messages.TAG_ALREADY_EXISTS;
+import static com.epam.esm.exceptions.Messages.TAG_ID_NOT_FOUND;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.springframework.http.HttpStatus.CREATED;
+import static org.springframework.http.HttpStatus.FOUND;
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+
 import com.epam.esm.Dto.Errors.ErrorDTO;
 import com.epam.esm.model.Tag;
-import com.epam.esm.repository.GiftCertificateTagRepository;
+import com.epam.esm.repository.TagRepository;
 import com.epam.esm.service.TagService;
-import com.epam.esm.Main;
+import java.util.Optional;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.InjectMocks;
 import org.mockito.Mock;
 import org.mockito.Mockito;
 import org.mockito.junit.jupiter.MockitoExtension;
-import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
-import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.http.ResponseEntity;
 
-import static com.epam.esm.exceptions.Codes.TAG_BAD_REQUEST;
-import static com.epam.esm.exceptions.Codes.TAG_NOT_FOUND;
-import static com.epam.esm.exceptions.Messages.*;
-import static org.junit.jupiter.api.Assertions.*;
-import static org.springframework.http.HttpStatus.*;
-
 @ExtendWith(MockitoExtension.class)
 public class TagServiceTest {
 
     public static final String TAG_NAME = "name";
-    public static final long TAG_ID = 1L;
+    public static final Long TAG_ID = 1L;
 
     @InjectMocks
     private TagService tagService;
 
     @Mock
-    GiftCertificateTagRepository giftCertificateTagRepository;
+    TagRepository tagRepository;
 
     @Mock
     Tag tag;
 
     @Test
     public void saveTag_savedCorrectly_returnCreated() {
 
-        Mockito.when(giftCertificateTagRepository.saveTag(TAG_NAME)).thenReturn(tag);
-        Mockito.when(giftCertificateTagRepository.getTagById(TAG_ID)).thenReturn(tag);
-        Mockito.when(tag.getId()).thenReturn(TAG_ID);
+        Mockito.when(tagRepository.save(any(Tag.class))).thenReturn(tag);
+        Mockito.when(tagRepository.findByName(TAG_NAME)).thenReturn(Optional.empty());
 
         ResponseEntity<?> actual = tagService.saveTag(TAG_NAME);
-        Mockito.verify(giftCertificateTagRepository).saveTag(Mockito.eq(TAG_NAME));
+        Mockito.verify(tagRepository).save(any(Tag.class));
 
         assertEquals(CREATED, actual.getStatusCode());
         assertEquals(tag, actual.getBody());
     }
 
-    @Test
-    public void saveTag_cannotSaveTag_returnBadRequest() {
```

Comment:
As I've commented about JPA exceptions in the TagService itself, here you would need to test that exception part and make sure you properly catch and process exception from repository. Probably by imitating it via when(...).thanThrow(...)

Reply:
I will remake every test, so will take that into account.